src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMBooster.scala

```diff
@@ -175,16 +194,34 @@ class LightGBMBooster(val model: String) extends Serializable {
       if (raw) boosterHandler.rawScoreConstant
       else boosterHandler.normalScoreConstant
     features match {
-      case dense: DenseVector => predictScoreForMat(dense.toArray, kind, classification)
-      case sparse: SparseVector => predictScoreForCSR(sparse, kind, classification)
+      case dense: DenseVector => predictForMat(dense.toArray, kind,
+        boosterHandler.scoredDataLengthLongPtr, boosterHandler.scoredDataOutPtr)
```

Comment:
I think you need to do .get().ptr here?  and in other places as well

Reply:
yes, I forgot to change that